src/core/service_model.jl

```diff
@@ -85,6 +88,13 @@ get_duals(m::ServiceModel) = m.duals
 get_time_series_names(m::ServiceModel) = m.time_series_names
 get_attributes(m::ServiceModel) = m.attributes
 get_attribute(m::ServiceModel, key::String) = get(m.attributes, key, nothing)
+get_contributing_devices_map(m::ServiceModel) = m.contributing_devices_map
+get_contributing_devices_map(m::ServiceModel, key) =
+    get(m.contributing_devices_map, key, nothing)
+get_contributing_devices(m::ServiceModel) =
```

Comment:
I don't understand this function. Are you trying to get all the contributing devices?

Reply:
Iterators.Flatten(values(m.contributing_devices_map))

Reply:
The downstream implementation of add_variables only take a `Vector{PSY.Component}` or  `IS.FlattenIteratorWrapper{PSY.Component}`. So leaving it as is for now

Reply:
Can you leave a comment in the code please